serialize.js

```diff
@@ -18,18 +22,21 @@ function serializeElement(elem) {
         tagname = tagname.toLowerCase()
     }
 
-    strings.push("<" + tagname +
-        properties(elem) + datasetify(elem) + ">")
+    strings.push("<" + tagname + properties(elem) + datasetify(elem))
 
-    if (elem.textContent || elem.innerText) {
-        strings.push(elem.textContent || elem.innerText)
-    }
+    if (voidElements.test(tagname)) {
+        strings.push(" />")
+    } else {
+        strings.push(">")
 
-    elem.childNodes.forEach(function (node) {
-        strings.push(serializeNode(node))
-    })
+        if (elem.childNodes.length) {
+            strings.push.apply(strings, elem.childNodes.map(serializeNode))
+        } else {
```

Comment:
lets keep this an else if elem.textContent elem.innerText;

This way we dont push an empty string into the strings array.